gateway/hostname.go

```diff
@@ -306,22 +306,59 @@ func isHTTPSRequest(r *http.Request) bool {
 
 // Converts a FQDN to DNS-safe representation that fits in 63 characters:
 // my.v-long.example.com → my-v--long-example-com
-func toDNSLinkDNSLabel(fqdn string) (dnsLabel string, err error) {
+// Implements specification from https://specs.ipfs.tech/http-gateways/subdomain-gateway/#host-request-header
+func InlineDNSLink(fqdn string) (dnsLabel string, err error) {
```

Comment:
```suggestion
// InlineDNSLink implements specification from https://specs.ipfs.tech/http-gateways/subdomain-gateway/#host-request-header
func InlineDNSLink(fqdn string) (dnsLabel string, err error) {
```

Just Go convention-stuff.